### ISSUE TITLE
Incorrectly reported missing quotes in yaml file when used in molecule 3.2.2

### ISSUE THREAD

**Issue by JohnMalmberg** [ansible-community/ansible-lint#1379]
##### Summary
Tried to use newest version of ansible-lint in molecule and it is claiming that quotes are missing when they are present.

I do not get this exact misbehavior when run outside of molecule.

Changing ansble-lint back to 4.3.7 results in the expected behavior for molecule runs.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.15
ansible-lint --version
ansible-lint 5.0.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Python 3.6 virtuialenv on CentOS 7.9.

cat requirements.txt
ansible==2.9.15
ansible-lint==5.0.1
molecule==3.2.2
molecule-docker
yamllint
ansible-vagrant
docker

##### STEPS TO REPRODUCE

```console (paste below)
cat molecule/default/converge.yml
---
- name: Converge
  hosts: all
  become: yes
  gather_facts: yes

  # https://github.com/ansible-community/molecule/issues/1567
  roles:
    - role: "{{ lookup('env', 'MOLECULE_PROJECT_DIRECTORY') | basename }}"

# Issue the command
molecule test
```

##### Desired Behaviour

It should pass lint as it does in ansible-lint 4.3.7

Changing ansble-lint back to 4.3.7 results in the expected behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behaviour

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below
Loading custom .yamllint config file, this extends our internal yamllint config.
WARNING  Listing 1 violation(s) that are fatal
internal-error: the role '.my_role' was not found in /home/mypath/roles/my_role/molecule/default/roles:/home/myhome/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/mypath/roles/myrole/molecule/default
molecule/default/converge.yml:37:7 [WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that
the implicit localhost does not match 'all'
ERROR! the role 'myrole' was not found in /home/mypath/roles/do.snapshot_host/molecule/default/roles:/home/myhome/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/mypath/roles/myrole/molecule/default


The error appears to be in '/home/mypath/roles/do.snapshot_host/molecule/default/converge.yml': line 37, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  roles:
    - role: "{{ lookup('env', 'MOLECULE_PROJECT_DIRECTORY') | basename }}"
      ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"

```

**Comment by ssbarnea** [comment]
That is not a supported use-case as it ansible-playbook --syntax-check will not be able to pass on these files either.  As the maintainer of molecule itself I would also need to advice against such tricks as they rely on molecule internals, just mention role name there, as testing playbooks should be as close as the productions ones as possible.

The error you are seeing comes from ansible itself not the linter. Please try to use the discussions when not sure if about the bug. Please note that I will be more happy to review a patch on molecule that makes it easier to find the role being tested by its name. Keep in mind that ideally converge.yml should be a playbooks that people can execute even without molecule itself.
